sopel/plugins/__init__.py

```diff
@@ -67,12 +67,15 @@ def find_internal_plugins():
     Internal plugins can be found under ``sopel.modules``. This list does not
     include the ``coretasks`` plugin.
     """
-    plugin_dir = imp.find_module(
-        'modules',
-        [imp.find_module('sopel')[1]]
-    )[1]
-
-    for name, _ in _list_plugin_filenames(plugin_dir):
+    modules = importlib.util.find_spec('sopel.modules')
+    plugin_list = itertools.chain(
+        itertools.chain.from_iterable([
+            _list_plugin_filenames(path)
+            for path in modules.submodule_search_locations
+        ])
+    )
```

Comment:
You don't need the first `itertools.chain`, as you give it only one iterable. And you don't need to use a list-comprehension: a generator-expression is fine, as `itertools.chain.from_iterable` accepts any iterable:

```suggestion
    plugin_list = itertools.chain.from_iterable(
        _list_plugin_filenames(path)
        for path in modules.submodule_search_locations
    )
```